verto/processor-info.json

```diff
@@ -158,8 +158,23 @@
           }
         },
         "template_parameters": {
+          "full_file_path": {
+            "argument": "full_file_path"
+          },
           "file_path": {
-            "argument": "file-path"
+            "argument": "file_path"
+          },
+          "file_width_value": {
+            "argument": "file_width_value"
+          },
+          "file_width_unit": {
+            "argument": "file_width_unit"
+          },
+          "file_extension": {
+            "argument": "file_extension"
+          },
+          "file_relative": {
+            "argument": "file_relative"
```

Comment:
`caption_link` should be changed to `caption-link` for consistency

Reply:
This was something that did my head in for ages (and still does, hope it's all correct) so will try to explain it as best as possible. Here we go...

In `processor-info.json`, there is two major sections used for `image-tag` and `image-container`:

1. `arguments` where all the keys must use dashes for readability to authors when writing.
2. `template_parameters` where all the keys must uses underscores as these need to be valid Python variable names for the templating engine.

For each key in `template_parameters` there is an key called `argument` which refers to the key to retrieve the value from a dictionary. It appears all the values in the `arguments` go directly into this dictionary, but can be modified in the dictionary within the processor or modified in the JSON file (as on line 15).

Most of the `argument` keys in the `template_parameters` section for the `image-tag` and `image-container` contain the value with the dash, as it's directly taking the value from the `arguments` section into the template. For example, `caption-link` and `hover-text` are taken into the template as is. Hopefully that makes sense, but it's about to get more complicated...

I had to create a utility function `image_file_name_components` to split the filename into the many pieces I needed (extension, width value, width unit, etc), and it loads all those values into a dictionary. Originally this dictionary had all it's keys with dashes instead of underscores, and it was directly merged into the context dictionary for template rendering, therefore all the `argument`s under `template_parameters` in the JSON had dashes for consistency.

Then comes in the young whippersnapper `image-inline` that has to be different. It's an inline pattern defined as a custom processor. If you look at it's definition in the JSON file, you will see that it has `arguments` for the tag, but no `template_parameters` as the processor directly modifies the HTML tree within the processor.

I tried to use my utility function to send the same dictionary for rendering the template of `image-inline` but was getting errors that the keys weren't valid Python variable names, because they contained dashes and not underscores. This is because it contain the middle step like the other two image processors where the variables are switched to values with dashes.

Therefore, I changed the utility function to return keys with underscores instead of dashes so `image-inline` could use the function correctly, and changed the JSON file for `image-tag` and `image-container` to look for those particular attributes with underscores (hence why line 244 is looking for a value with underscores, and line 262 is looking for one with dashes).

Overall, it means:

- All tag arguments are dashed for authors.
- All template tags are underscored for developers.
- The only place where mixed cased is used is within Verto's JSON file. This seemed better than writing the utility function to sometimes return with dashes or underscores (but it can be done if preferred).

Phew! That took a lot of words. Here's a prize for making it through :pineapple: 

Let me know if you have any questions about it.

Reply:
I actually wondered if this was the reason it was done, so fine by me 🍍